### ISSUE TITLE
[Wish] Idea: network processing client

### ISSUE THREAD

**Issue by natowi** [alicevision/meshroom#357]
A simple Meshroom client for network processing to distribute the workload on multiple machines would be useful.

The idea is to utilize the computers you have at home/in the office/school/university/... 
This could be done by directly connecting running Meshroom instances and set one as master and all the others as computing clients or by installing MR computing clients and connect them to MR.

The installation and setup should be as simple as possible.

(Not everyone has access to a renderfarm/workstation)

**Comment by fabiencastan** [comment]
That was the idea of PR #81, without re-inventing the wheel. Fireworks allows to launch a master and slaves on other machines easily.
But I never managed to finalize it...

**Comment by fabiencastan** [comment]
If you can test fireworks, I would love to have other feedback to decide if it's the right way to go!
From my test, it was easy to install and setup (I have only tested on linux for now). The UI for monitoring is limited, but as we have progression feedback within meshroom, that's not a too big issue.

**Comment by natowi** [comment]
@fabiencastan Thank you. I installed fireworks and added the fireworks submitter to Meshroom. However Meshroom does not find the fireworks installation. Do I have to add/edit a settings file somewhere to point Meshroom to fireworks? I'll give it another try in a few weeks when I have more time.

**Comment by ShalokShalom** [comment]
Interested in this

**Comment by natowi** [comment]
Where do I have to install fireworks to allow this submitter to find  the fireworks module?
I get the error WARNING:root:== The following plugins could not be loaded ==
  * fireworksSubmitter: No module named 'fireworks'
  * simpleFarmSubmitter: No module named 'simpleFarm'

**Comment by thegallivanter** [comment]
Natowi did you ever get distributed processing with Fireworks set up and working?

**Comment by natowi** [comment]
@thegallivanter No, I did not figure out how to link Meshroom and fireworks, but I will continue looking into this. This should not be too difficult, but without an (working) example it is time consuming to figure out. This is on the to-do list for the documentation.

**Comment by natowi** [comment]
Reference https://github.com/alicevision/meshroom/pull/213

**Comment by natowi** [comment]
@sergiy-nazarenko I [read ](https://github.com/alicevision/meshroom/issues/659) you successfully use meshroom on a render farm. Can you describe your render farm setup? The documentation (wiki) on this feature is quite thin and could use some input.

**Comment by sergiy-nazarenko** [comment]
Hej, natowi
Sure, I can describe. I actually modified the simpleFarmSubmitter.py , here is the link of our slurmFarmSubmitter :
https://raw.githubusercontent.com/sergiy-nazarenko/hafarm/0.3.3/scripts/meshroom/submitters/slurmFarmSubmitter.py
We use slurm on render farm. Slurm needs bash script to load jobs with specific comments such #SBATCH . slurmFarmSubmitter converts the meshroom nodes to those scripts through python + jinja template
https://github.com/sergiy-nazarenko/hafarm/blob/master/slurm_job.schema

Cheers,
Sergiy

**Comment by spreck** [comment]
@natowi @sergiy-nazarenko 
Is there any more info on how to implement this?

I looked through the linked documents, but frankly they are beyond my level of expertise. If this could be made more easily accessible that would be great. Any tips on how to go about using multiple computers for meshroom reconstructions?

**Comment by natowi** [comment]
@spreck https://github.com/alicevision/meshroom/wiki/Renderfarm-submitters
